core-overlay.html

```diff
@@ -111,666 +124,373 @@
-      }
-    },
+        this.applyFocus();
 
-    completeBackdrop: function() {
-      if (this.backdrop) {
-        trackBackdrop(this);
-        if (getBackdrops().length === 0) {
-          this.scrim.parentNode.removeChild(this.scrim);
-        }
-      }
-    },
+        // complete all tasks before firing -completed event
+        this.fire('core-overlay-' + (this.opened ? 'open' : 'close') +
+            '-completed');
+      },
 
-    preparePositioning: function() {
-      this.target.style.transition = this.target.style.webkitTransition = 'none';
-      this.target.style.transform = this.target.style.webkitTransform = 'none';
-      this.target.style.display = '';
-    },
+      prepareBackdrop: function() {
+        if (this.backdrop && this.opened) {
+          if (!this._backdrop.parentNode) {
+            Polymer.dom(document.body).appendChild(this._backdrop);
+            this._backdrop.style.zIndex = manager.currentOverlayZ() - 1;
+          }
+          manager.trackBackdrop(this);
+        }
+      },
 
-    discoverDimensions: function() {
-      if (this.dimensions) {
-        return;
-      }
-      var target = getComputedStyle(this.target);
-      var sizer = getComputedStyle(this.sizingTarget);
-      this.dimensions = {
-        position: {
-          v: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
-            'bottom' : null),
-          h: target.left !== 'auto' ? 'left' : (target.right !== 'auto' ?
-            'right' : null),
-          css: target.position
-        },
-        size: {
-          v: sizer.maxHeight !== 'none',
-          h: sizer.maxWidth !== 'none'
-        },
-        margin: {
-          top: parseInt(target.marginTop) || 0,
-          right: parseInt(target.marginRight) || 0,
-          bottom: parseInt(target.marginBottom) || 0,
-          left: parseInt(target.marginLeft) || 0
+      renderBackdropOpened: function() {
+        if (this.backdrop && manager.getBackdrops().length < 2) {
+          this._backdrop.classList.toggle('core-opened', this.opened);
         }
-      };
-    },
-
-    finishPositioning: function(target) {
-      this.target.style.display = 'none';
-      this.target.style.transform = this.target.style.webkitTransform = '';
-      // force layout to avoid application of transform
-      this.target.offsetWidth;
-      this.target.style.transition = this.target.style.webkitTransition = '';
-    },
-
-    getTransition: function(name) {
-      return this.meta.byId(name || this.transition);
-    },
-
-    getFocusNode: function() {
-      return this.target.querySelector('[autofocus]') || this.target;
-    },
-
-    applyFocus: function() {
-      var focusNode = this.getFocusNode();
-      if (this.opened) {
-        if (!this.autoFocusDisabled) {
-          focusNode.focus();
+      },
+
+      completeBackdrop: function() {
+        if (this.backdrop) {
+          manager.trackBackdrop(this);
```

Comment:
This is the second place I've seen this call to `manager.trackBackdrop`. Is the method intended to be idempotent?